### PR TITLE
<FEAT>: Update Requirements files

### LINKED ISSUE #121: [Enhc] Split library requirements into dev and user requirements
### Requested Feature/Enhancement

Create separate LDAR-Sim requirements installs for developers and for users. 

### Need for feature/Enhancement

This way users who have no plans of contributing to LDAR-Sim won't need to install developer requirements such as testing libraries.

### Problem to solve

_No response_

### Implementation Ideas

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SEAJang**
Reason for change:
Requirements files are outdated

Resolution:
Update the requirements files

Effect(s) of change:
Requirements files have been updated
In addition, dev specific requirements file has been added.

#  Update Requirements files

## What was changed

Update the file used to install LDAR-sim requirements, and add a developer specific requirements file.

## Intended Purpose

Update the installation files

## Level of version change required

N/A

## Testing Completed

Both requirement files were used to run LDAR-sim demo sim with no errors, in addition, the developer requirements were able to run both the e2e tests and unit tests successfully. Files tested using both windows OS and Linux. 

[test_results.zip](https://github.com/LDAR-Sim/LDAR_Sim/files/13055895/test_results.zip)

## Target Issue

Closes #121 .

## Additional Information

Include any other important information here.

### CODE REVIEW

File: LDAR_Sim/install/requirements.txt [hunk 1]
```diff
@@ -31,23 +30,20 @@ packaging==23.1
...
 pycparser==2.21
 pyOpenSSL==23.1.1
-pyparsing==3.0.9
-PyQt5-sip==12.11.0
-PySocks==1.7.1
-python==3.11.3
+pyparsing
+# PyQt5-sip==12.11.0
+# PySocks==1.7.1
+#python==3.11.0
```
Comment (ThomasGalesloot) [inline LDAR_Sim/install/requirements.txt:45]: Question: Why have these been left in but commented out?

File: LDAR_Sim/install/requirements_dev.txt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ephem==4.1.4
+fonttools==4.39.3
+hypothesis==6.87.0
+idna==3.4
+jmespath==1.0.1
+joblib==1.2.0
+jsonpatch==1.32
+jsonpointer==2.0
+kiwisolver==1.4.4
+matplotlib==3.7.1
```
Comment (ThomasGalesloot) [inline LDAR_Sim/install/requirements_dev.txt:25]: Question: Why is matplotlib dev only?

**Review by ThomasGalesloot — APPROVED** [review]
A few questions here: Some of the requirements in the dev requirements seems like they are unnecessary or perhaps dependencies for used libraries but unused themselves. In order to keep the requirements clean could they be trimmed out? Second have all the required libraries been updated to their latest version?

File: LDAR_Sim/install/requirements.txt (hunk 1 above)
Comment (SEAJang) [reply LDAR_Sim/install/requirements.txt:43]: Mostly as reference. Will remove, all but the python.

**Pull request merged**